File: sdks/java/src/test/java/com/autlantic/billing/ClientTest.java
package com.autlantic.billing;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Map;
import org.junit.jupiter.api.Test;

final class ClientTest {
  @Test
  void fromEnvRequiresKey() {
    assertThrows(
        AutlanticBillingException.class, () -> AutlanticBilling.fromEnv(Map.of()));
  }

  @Test
  void modeFromKey() {
    assertEquals("test", AutlanticBilling.billingModeFromApiKey("abk_test_x"));
    assertEquals("live", AutlanticBilling.billingModeFromApiKey("abk_live_x"));
  }

  @Test
  void constructPinsVersionConstants() {
    AutlanticBilling client = new AutlanticBilling("abk_test_demo");
    assertEquals("test", client.getMode());
    assertEquals("0.1.0", Version.SDK_VERSION);
    assertEquals("2026-01-01", Version.AUTLANTIC_API_VERSION);
    assertTrue(Version.USER_AGENT.contains("autlantic-java/"));
  }
}

File: sdks/php/tests/ClientTest.php
<?php

declare(strict_types=1);

namespace Autlantic\Billing\Tests;

use Autlantic\Billing\AutlanticBilling;
use Autlantic\Billing\AutlanticBillingException;
use Autlantic\Billing\Version;
use PHPUnit\Framework\TestCase;

final class ClientTest extends TestCase
{
    public function testFromEnvRequiresKey(): void
    {
        $this->expectException(AutlanticBillingException::class);
        AutlanticBilling::fromEnv([]);
    }

    public function testModeFromKey(): void
    {
        self::assertSame('test', AutlanticBilling::billingModeFromApiKey('abk_test_x'));
        self::assertSame('live', AutlanticBilling::billingModeFromApiKey('abk_live_x'));
    }

    public function testConstructPinsVersionConstants(): void
    {
        $client = new AutlanticBilling('abk_test_demo');
        self::assertSame('test', $client->mode);
        self::assertSame('0.1.0', Version::SDK_VERSION);
        self::assertSame('2026-01-01', Version::AUTLANTIC_API_VERSION);
        self::assertStringContainsString('autlantic-php/', Version::USER_AGENT);
    }
}
